backend/pkg/modules/internal/differ/diffs_sender.go

```diff
@@ -0,0 +1,108 @@
+package differ
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/openclarity/apiclarity/api3/common"
+	"github.com/openclarity/apiclarity/api3/global"
+	"github.com/openclarity/apiclarity/api3/notifications"
+	"github.com/openclarity/apiclarity/backend/pkg/database"
+)
+
+func (p *differ) StartDiffsSender(ctx context.Context) {
+	// each period aggregate diffs per api and notify to notification server
+	log.Info("Starting diffs sender")
+	interval := p.config.SendNotificationIntervalSec()
+
+	ticker := time.NewTicker(time.Duration(interval) * time.Second)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+			if err := p.sendDiffsNotification(); err != nil {
+				log.Errorf("Failed to send diffs notification. total diffs=%v.: %v", p.totalDiffs, err)
+			}
+			p.clearDiffs()
+		}
+	}
+}
+
+func (p *differ) clearDiffs() {
+	p.Lock()
+	defer p.Unlock()
+	p.apiIDToDiffs = map[uint]map[diffHash]global.Diff{}
+	p.totalDiffs = 0
+}
+
+func (p *differ) sendDiffsNotification() error {
+	if p.getTotalEvents() == 0 {
+		log.Infof("No events to send")
+		return nil
+	}
+
+	diffsNotification := p.getSpecDiffsNotification()
+
+	log.Infof("Sending diff notification: %+v", diffsNotification)
+
+	n := notifications.APIClarityNotification{}
+	if err := n.FromSpecDiffsNotification(diffsNotification); err != nil {
+		return err
+	}
+	if err := p.accessor.Notify(context.TODO(), moduleName, 0, n); err != nil {
```

Comment:
because it is not relevant (I send notification for several apis)

Reply:
If the notification path is `/notification/{apiID}` so I think it is kind of an abuse to send notification to `0` apiID with the real api id inside each notification. DYT?

Reply:
I think this is the general api that is defined for modules notification, but here this is not really fit. since we are also the server I dont see a problem with that

Reply:
decided to send two events